lib/class-wp-rest-block-types-controller.php

```diff
@@ -0,0 +1,359 @@
+	}
+
+	/**
+	 * Retrieves a specific block type.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$block_name = sprintf( '%s/%s', $request['namespace'], $request['name'] );
+		$block_type = $this->get_block( $block_name );
+		if ( is_wp_error( $block_type ) ) {
+			return $block_type;
+		}
+		$data = $this->prepare_item_for_response( $block_type, $request );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepares a block type object for serialization.
+	 *
+	 * @param WP_Block_Type   $block_type block type data.
+	 * @param WP_REST_Request $request    Full details about the request.
+	 *
+	 * @return WP_REST_Response block type data.
+	 */
+	public function prepare_item_for_response( $block_type, $request ) {
+
+		$fields = $this->get_fields_for_response( $request );
+		$data   = array();
+
+		if ( in_array( 'attributes', $fields, true ) ) {
+			$data['attributes'] = $block_type->get_attributes();
+		}
+
+		if ( in_array( 'is_dynamic', $fields, true ) ) {
+			$data['is_dynamic'] = $block_type->is_dynamic();
+		}
+
+		$schema       = $this->get_item_schema();
+		$extra_fields = array(
+			'name'         => 'name',
+			'editorScript' => 'editor_script',
+			'script'       => 'script',
+			'editorStyle'  => 'editor_style',
+			'style'        => 'style'
+		);
+		foreach ( $extra_fields as $key => $extra_field ) {
+			if ( in_array( $key, $fields, true ) ) {
+				$data[ $key ] = rest_sanitize_value_from_schema( $block_type->$extra_field, $schema['properties'][ $key ] );
+			}
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		$response->add_links( $this->prepare_links( $block_type ) );
+
+		/**
+		 * Filters a block type returned from the REST API.
+		 *
+		 * Allows modification of the block type data right before it is returned.
+		 *
+		 * @param WP_REST_Response $response   The response object.
+		 * @param object           $block_type The original block type object.
+		 * @param WP_REST_Request  $request    Request used to generate the response.
+		 */
+		return apply_filters( 'rest_prepare_block_type', $response, $block_type, $request );
+	}
+
+	/**
+	 * Prepares links for the request.
+	 *
+	 * @param WP_Block_Type $block_type block type data.
+	 * @return array Links for the given block type.
+	 */
+	protected function prepare_links( $block_type ) {
+		$links = array(
+			'collection' => array(
+				'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
+			),
+			'about'      => array(
+				'href' => rest_url( sprintf( '%s/%s/%s', $this->namespace, $this->rest_base, $block_type->name ) ),
+			),
```

Comment:
@TimothyBJacobs What you thoughts on adding a link to the render block api if this is a dynamic block?

Reply:
From my perspective, more links the better.

Reply:
The render link requires ?context=edit. Should this be included in the link?

Reply:
I would yes.

Reply:
Add render and namespace in https://github.com/WordPress/gutenberg/pull/21065/commits/8f32a723e817486be8bcc390605458feee101b60